app-attach/namespace.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+
+	"git.fd.io/govpp.git/api"
+	"github.com/edwarnicke/log"
+	"github.com/edwarnicke/vpphelper"
+)
+var doOnce sync.Once
+type Connection interface {
+	api.Connection
+	api.ChannelProvider
+}
+
+type Namespace struct {
+    vppConn vpphelper.Connection
```

Comment:
Golang supports [Duck Typing](https://medium.com/@matryer/golang-advent-calendar-day-one-duck-typing-a513aaed544d).

[vpphelper.Connection](https://github.com/edwarnicke/vpphelper/blob/3e6797de32c36a2eac2ac30e0932d04dc796fc6d/start.go#L36-L39) is just

```go
// Connection Combination of api.Connection and api.ChannelProvider
type Connection interface {
	api.Connection
	api.ChannelProvider
}
```

I would recommend you define your own 'Connection' similarly locally to your package.

The reason for this is that your eventual users may not be vpphelper users, and should not be forced to include vpphelper to use your library code.